src/main/kotlin/org/dif/message/Message.kt

```diff
@@ -0,0 +1,110 @@
+package org.dif.message
+
+import org.dif.common.JSON
+import org.dif.common.Typ
+
+data class Message(
+    val id: String,
+    val payload: JSON,
```

Comment:
Currently `JSON` can be a `JSONArray` (in this class hierarchy). But according to JWM spec and https://github.com/decentralized-identity/didcomm-messaging/issues/266 it can be a JSON Object only.